Tests/Kernel/TestPosixFallocate.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022, Andreas Kling <kling@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/System.h>
+#include <LibTest/TestCase.h>
+
+TEST_CASE(PosixFallocateBasics)
+{
+    char pattern[] = "/tmp/posix_fallocate.XXXXXX";
+    auto fd = MUST(Core::System::mkstemp(pattern));
+    VERIFY(fd >= 0);
+
+    {
+        auto result = Core::System::posix_fallocate(fd, 0, 1024);
+        EXPECT_EQ(result.is_error(), false);
+
+        auto stat = MUST(Core::System::fstat(fd));
+        EXPECT_EQ(stat.st_size, 1024);
+    }
+
+    {
+        auto result = Core::System::posix_fallocate(-1, 0, 1024);
+        EXPECT_EQ(result.is_error(), true);
+        EXPECT_EQ(result.error().code(), EBADF);
+    }
+
+    {
+        auto result = Core::System::posix_fallocate(fd, 0, -1);
+        EXPECT_EQ(result.is_error(), true);
+        EXPECT_EQ(result.error().code(), EINVAL);
+    }
+
+    {
+        auto result = Core::System::posix_fallocate(fd, 0, 0);
+        EXPECT_EQ(result.is_error(), true);
+        EXPECT_EQ(result.error().code(), EINVAL);
+    }
+
+    {
+        auto result = Core::System::posix_fallocate(fd, -1, 0);
+        EXPECT_EQ(result.is_error(), true);
+        EXPECT_EQ(result.error().code(), EINVAL);
+    }
+}
+
+TEST_CASE(FallocateOnDeviceFile)
```

Comment:
```suggestion
TEST_CASE(PosixFallocateOnDeviceFile)
```

Reply:
Should probably also be snake_case, most (if not all) of our current test cases are snake_case instead of CamelCase.